pkg/ebpf/controlplane/controller.go

```diff
@@ -0,0 +1,161 @@
+package controlplane
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/aquasecurity/libbpfgo"
+
+	"github.com/aquasecurity/tracee/pkg/capabilities"
+	"github.com/aquasecurity/tracee/pkg/containers"
+	"github.com/aquasecurity/tracee/pkg/ebpf/probes"
+	"github.com/aquasecurity/tracee/pkg/errfmt"
+	"github.com/aquasecurity/tracee/pkg/events"
+	"github.com/aquasecurity/tracee/pkg/events/parse"
+	"github.com/aquasecurity/tracee/pkg/logger"
+	"github.com/aquasecurity/tracee/types/trace"
+)
+
+const pollTimeout int = 300 // from tracee.go (move to a consts package?)
+
+// Control Plane probe handles
+const (
+	cgroupMkdirControlProbe probes.Handle = iota
+	cgroupRmdirControlProbe
+)
+
+type Controller struct {
+	ctx            context.Context
+	signalChan     chan []byte
+	lostSignalChan chan uint64
+	bpfModule      *libbpfgo.Module
+	probeGroup     *probes.ProbeGroup
+	signalBuffer   *libbpfgo.PerfBuffer
+	cgroupManager  *containers.Containers
```

Comment:
Curious why "cgroupManager" ? Didnt follow the "manager" wording.

Reply:
I thought "is it something worth changing in Tracee singletoon as well (instead of "containers" only, to have something related to "manager", depending on your thoughts).

Reply:
Because originally I defined an internal interface which the `Containers` struct fulfills which had `CgroupMkdir` and the `RemoveFromBPFMap` and `CgroupRmdir`. I eventually opted not to use it, but I may add it back for unit tests later. That interface, in terms of naming fit som thing like `cgroupManager` best.

About changing in the tracee singleton, I don't know honestly. The `containers` struct right now kinda does two job which is tracking cgroups and interpreting them as containers, which i'm not sure can be separated elegantly, so _in that context_, changing to `cgroupsManager` or `containerManager` doesn't seem right.